src/chain/block.cpp

```diff
@@ -120,7 +118,7 @@ block& block::operator=(block&& other)
     non_coinbase_inputs_ = other.non_coinbase_inputs_cache();
     header_ = std::move(other.header_);
     transactions_ = std::move(other.transactions_);
-    metadata = std::move(other.metadata);
+    metadata = other.metadata;
```

Comment:
Please clarify why assignment is preferred.

Reply:
>As a matter of style we never use const on non-ref parameters.

Done.

>Please clarify why assignment is preferred.

Because metadata's type is trivially-copyable and the std::move has no effect. That is the reason I decided to remove std::move. In addition, I've added a new unit test to see if the type stays trivially-copyable. 

The question is whether to use it preventively if the implementation of metadata's type changes. Moreover, one can argue in the following way, std::move belongs in the body of a move assignment operator, no matter if it brings something, to show the explicitly intent.

Similar question:
https://stackoverflow.com/questions/49241201/is-it-undesirable-to-defensively-apply-stdmove-to-trivially-copyable-types#

Reply:
Thanks, that's what I figured, just wanted to make sure I wasn't missing something.

Clearly we wouldn't use `std::move` in the case of a integral data type that always copied, as that is self-documenting and guaranteed. But in the case of other types (which are always subject to internal change) the simplest approach is to always use `std::move` explicitly. This serves to make the code self-documenting (no need to search for the external guard, such as a unit test), and eliminates the additional cost of implementing and maintaining the external guard.

Also I think this is a pattern implemented throughout the codebase, so it's a question of consistent style as well.

Reply:
Done.